dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1167,22 +1170,44 @@ private IHTMLPage loadPageByIdentifier ( String ident, boolean live, User user,
 
         final List<MultiTree> trees = APILocator.getMultiTreeAPI().getMultiTreesByChild(id.getId());
         for (final MultiTree tree : trees) {
-            final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
-                    .findByIdLanguageFallback(tree.getParent1(), contentlet.getLanguageId(), false,
-                            APILocator.getUserAPI().getSystemUser(), false);
-            final Container container = APILocator.getContainerAPI()
-                    .getWorkingContainerById(tree.getParent2(),
-                            APILocator.getUserAPI().getSystemUser(), false);
-            if (InodeUtils.isSet(page.getInode()) && InodeUtils.isSet(container.getInode())) {
-                final Map<String, Object> map = new HashMap<String, Object>();
-                map.put("page", page);
-                map.put("container", container);
-                results.add(map);
+
+            try {
+                final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
+                        .findByIdLanguageFallback(tree.getHtmlPage(), contentlet.getLanguageId(),
+                                false,
+                                APILocator.getUserAPI().getSystemUser(), false);
+
+                final Container container = APILocator.getContainerAPI()
+                        .getWorkingContainerById(tree.getContainer(),
+                                APILocator.getUserAPI().getSystemUser(), false);
+
+                if (InodeUtils.isSet(page.getInode()) && InodeUtils.isSet(container.getInode())) {
+                    final String personaName = getPersonaNameByMultitree(tree);
+                    final Map<String, Object> map = new HashMap<>();
+                    map.put("page", page);
+                    map.put("container", container);
+                    map.put("persona", personaName);
+                    results.add(map);
+                }
+            } catch(DoesNotExistException e) {
+                Logger.debug(this, "Page not available in the requested language. This is ok");
             }
         }
         return results;
     }
 
+    private String getPersonaNameByMultitree(final MultiTree tree) throws DotSecurityException, DotDataException {
+        final String defaultPersona = Try.of(()->LanguageUtil.get(DEFAULT_PERSONA_NAME_KEY)).getOrElse("Default Visitor");
+        String personaTag = Try.of(()-> tree.getPersonalization()
```

Comment:
I think here you can return null,

if null return the defaultPersona.

I would also do the defaultPersona as a supplier to avoid the call if not needed

Reply:
Done